video/img_format.c

```diff
@@ -127,6 +127,20 @@ static const struct mp_imgfmt_entry mp_imgfmt_list[] = {
     [IMGFMT_VAAPI - IMGFMT_CUST_BASE] = {
         .name = "vaapi",
     },
+    [IMGFMT_RPI8 - IMGFMT_CUST_BASE] = {
+        .name = "rpi4_8",
+        .desc = { .flags = MP_IMGFLAG_COLOR_YUV,
+            .chroma_xs = 1, .chroma_ys = 1,
+            .comps = { {0, 0, 8}, {1, 0, 8}, {1, 1*8, 8}, {0} },
+        },
+    },
+    [IMGFMT_RPI10 - IMGFMT_CUST_BASE] = {
+        .name = "rpi4_10",
+        .desc = { .flags = MP_IMGFLAG_COLOR_YUV,
+            .chroma_xs = 1, .chroma_ys = 1,
+            .comps = { {0, 0, 8}, {1, 0, 8}, {1, 1*8, 8}, {0} },
+        },
+    },
```

Comment:
Without this some of the initial internal mpv checks fail, leading to ra_get_imgfmt_desc() failing. I tried to figure out another way to do it by mimicking something like VAAPI but I couldn't get it working. This is what I came up with instead.

Reply:
if that ffmpeg fork doesn't even define the `AV_PIX_FMT` constants then it probably doesn't define the descriptor either

Reply:
Yeah. This is one of the infamous "Why do we assert when trying to use SoC formats?" bugs. The descriptor has to be defined on the ffmpeg side. And @jeeb reminds us that there's a fate test which enforces this. Someone should run it...

Reply:
Seems like I mentioned `make fate-pixelutils` in a previous PR made for things to work better with one of the branches in that fork. https://github.com/mpv-player/mpv/pull/10344#issuecomment-1168989775

Generally speaking, at least now and then I would recommend anyone poking at FFmpeg to go through the FATE routine, the short version of which is as follows:
```bash
# configuration and compilation
../ffmpeg/configure --your --args
make

# this runs only the stuff which would fail if the descriptor was missing
make fate-pixelutils

# syncs the testing samples to the SAMPLES location you provide;
# the following is about 1.2G gigs or so looking at my dev VM, the samples are kept pretty minimal
make fate-rsync SAMPLES=../fate-suite/

# this runs the full suite
make fate SAMPLES=../fate-suite/
```

Reply:
@EmperorPenguin18 I don't want to make your life miserable, but it would much better if you could put the equivalent descriptors into an rpi-ffmpeg PR - that's a general improvement and then we don't need this work-around on the mpv side.

Reply:
And we only need an `HWACCEL` descriptor in ffmpeg, so you don't need to describe the format accurately (which isn't actually possible given how crazy the format is).

Reply:
Latest commit should resolve this.